integration/helpers.go

```diff
@@ -179,21 +182,44 @@ func ExpectEntriesInMultipleArrayToMatch[T any](output string, normalize func(*T
 	return expectEntriesToMatch(entries, expectedEntries...)
 }
 
-func BuildCommonData(namespace string) eventtypes.CommonData {
-	return eventtypes.CommonData{
-		Namespace: namespace,
-		// Pod and Container name are defined by BusyboxPodCommand.
-		Pod:       "test-pod",
-		Container: "test-pod",
+type CommonDataOption func(commonData *eventtypes.CommonData)
+
+// WithRuntimeMetadata sets the runtime and container name in the common data.
+// Notice the container name is taken from the Kubernetes metadata.
+func WithRuntimeMetadata(runtime string) CommonDataOption {
```

Comment:
Yep. It is because it is an option for `BuildCommonData`, which mainly creates a `CommonData` with k8s metadata:

https://github.com/inspektor-gadget/inspektor-gadget/blob/a39e7b00cc2d114259f3d9121aa4232ac7f6e407/integration/helpers.go#L196-L212

Instead, in the non-k8s tests, we manually create the `CommonData`. For example:

https://github.com/inspektor-gadget/inspektor-gadget/blob/a39e7b00cc2d114259f3d9121aa4232ac7f6e407/integration/ig/non-k8s/trace_dns_test.go#L76-L84

I agree `BuildCommonData` could be generalized using the options pattern to work for both cases, but, IMO, we can manage it in another PR. WDYT?

Reply:
> but, IMO, we can manage it in another PR. WDYT?

Yeah, We can handle such fixups in seperate PRs.